frontend/src/components/maps/MapNavBar.tsx

```diff
@@ -0,0 +1,33 @@
+import { useKeycloak } from '@react-keycloak/web';
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { Navbar, Nav, NavDropdown, Image } from 'react-bootstrap';
+import './MapNavBar.scss';
+import profileUrl from './profile.svg';
+
+/**
+ * Nav bar with with user-related functionality. TODO: potentially move up to the App.tsx level.
+ */
+function MapNavBar() {
+  const { keycloak } = useKeycloak();
+  const history = useHistory();
+  return (
+    <Navbar className="map-nav">
+      <Nav>
+        <Nav.Item className="profile">
+          <Image src={profileUrl} rounded/>
+        </Nav.Item>
+        <NavDropdown title={keycloak?.profile?.firstName} id="user-dropdown" >
+          <NavDropdown.Item onClick={() => {
+            history.push('/');
+            keycloak!!.logout();
```

Comment:
What's the purpose of the double exclamation mark here?

Reply:
this is the syntax for a forced de-reference in typescript. This will throw a null de-reference error if keycloak is null.

My thinking here was that keycloak should never be null at this point in the code, so throwing an exception at this point is acceptable as a null value would reflect a programming error of some kind.

Reply:
Could you please provide a link where a double ! is used? All I could find was the single "!" operator - named **non-null assertion** operator... 

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

In any case, static analysis tools like sonarqube suggests not to use it and instead do an explicit check for null or undefined. Thoughts?

https://rules.sonarsource.com/typescript/RSPEC-2966

Reply:
I think what I would end up doing here is explicitly checking for null and then throwing an exception myself. I'm not against that if I come up with a wrapper function for useKeycloak just so that I don't have to repeat this logic throughout the application.